app.js

```diff
@@ -1,4 +1,72 @@
 //  in this file create an express application - use the middle-ware built into express
 //  to serve up static files from the public directory (index.html and client.js - you 
 //  can also serve up css files from the public directory this way if you want)
-//  you need to support a '/trucks' endpoint, and a dynamic route for '/trucks:name'
\ No newline at end of file
+//  you need to support a '/trucks' endpoint, and a dynamic route for '/trucks:name'
+var _ = require('underscore');
+var express = require('express');
+var trucks = require('./trucks');
+var app = express();
+
+// to use the "static" middle-ware, we call it from our express object
+// this will allow us to serve up static files from "public"
+app.use(express.static('public'));
+
+app.get('/trucks', function (req, res) {
+	res.json(trucks.getTrucks());
+});
+
+app.get('/trucks/:name', function (req, res) {
+	var truck = req.params.name;
+	var truckInfo = trucks.getTruck(truck);
+	// Since some of our data is in arrays, it needs to be cleaned up
```

Comment:
Awesome, that's a good tip.  It's also a good reminder that I should check the available array methods more often, I always forget how many helpers like this there are.